src/Microsoft.VisualStudio.Services.Agent/Util/ApiUtil.cs

```diff
@@ -63,6 +64,7 @@ public static VssCredentials GetVssCredential(ServiceEndpoint serviceEndpoint)
             ArgUtil.NotNull(serviceEndpoint.Authorization, nameof(serviceEndpoint.Authorization));
             ArgUtil.NotNullOrEmpty(serviceEndpoint.Authorization.Scheme, nameof(serviceEndpoint.Authorization.Scheme));
 
+            WebProxy.ApplyProxySettings();
```

Comment:
Why is this needed in both places (CreateConnection and GetVssCredential)? Can you leave a comment in the source code that explains?